src/modinforegular.cpp

```diff
@@ -654,61 +653,60 @@ std::vector<ModInfo::EFlag> ModInfoRegular::getFlags() const
 }
 
 
-std::vector<ModInfo::EContent> ModInfoRegular::getContents() const
+std::vector<ModInfo::EContent> ModInfoRegular::doGetContents() const
 {
-  QTime now = QTime::currentTime();
-  if (m_LastContentCheck.isNull() || (m_LastContentCheck.secsTo(now) > 60)) {
-    m_CachedContent.clear();
-    QDir dir(absolutePath());
-    if (dir.entryList(QStringList() << "*.esp" << "*.esm" << "*.esl").size() > 0) {
-      m_CachedContent.push_back(CONTENT_PLUGIN);
-    }
-    if (dir.entryList(QStringList() << "*.bsa" << "*.ba2").size() > 0) {
-      m_CachedContent.push_back(CONTENT_BSA);
-    }
-    //use >1 for ini files since there is meta.ini in all mods already.
-    if (dir.entryList(QStringList() << "*.ini").size() > 1) {
-      m_CachedContent.push_back(CONTENT_INI);
-    }
+  auto tree = contentFileTree();
+  std::vector<ModInfo::EContent> contents;
 
-    if (dir.entryList(QStringList() << "*.modgroups").size() > 0) {
-      m_CachedContent.push_back(CONTENT_MODGROUP);
+  for (auto e : *tree) {
```

Comment:
There's some case issues:
 1) You specifically converting to lowercase for the suffix;
 2) `compare()` uses `FileNameComparator`, which is hardcoded to case insensitive (but doesn't actually use `CaseSensitivity`);
3) The `"dll"` check below is case sensitive (confusion between `FileTreeEntry::compare()` and `QString`?)

I also don't like `compare()`, it's three-way, so it's error-prone, and its name is unclear (compare what?) Really, this whole thing needs a complete refactor.

Reply:
The name `compare` comes from `QString`. I"m not against changing it but it means changing it everywhere (uibase, modorganizer, all the game plugins, all the installers, plugin python interface, etc.).

> 1. You specifically converting to lowercase for the suffix;

I just don't wanted to do `.compare("XX", FileNameComparator::CaseSensitivity)` everywhere. 

> 2. compare() uses FileNameComparator, which is hardcoded to case insensitive (but doesn't actually use CaseSensitivity);

That's an error on my side, `compare()` should use `CaseSensitivity`.

> 3. The "dll" check below is case sensitive (confusion between FileTreeEntry::compare() and QString?)

That's also an error on my side, it should be `compare("dll", FileNameComparator::CaseSensitivity)`.